src/python/pants/backend/docker/subsystems/docker_options.py

```diff
@@ -151,6 +151,16 @@ def env_vars(self) -> tuple[str, ...]:
             """
         ),
     )
+
+    enable_podman = BoolOption(
+        default=False,
+        help=softwrap(
+            """
+            EXPERIMENTAL. Allow support for podman where available instead of docker.
+            """
+        ),
+    )
```

Comment:
I kind of feel like registering this option as a plugin option from an experimental backend perhaps something like: `pants.backend.experimental.docker.podman`, named `experimental_enable_podman` and also default to `True` (so enabling the backend is all that you need to do to enable podman, while being experimental).
Once graduating out of experimental, we can deprecate the experimental option and backend, add this option you have here as the migration path, now still with default `False`. So once graduated it's just the `[docker].enable_podman` option you need to use.

All of the above just to isolate experimental features behind experimental backends as much as feasible.

Example for how to register plugin options: https://github.com/pantsbuild/pants/blob/952c4cb8479f1a855b85f3f2e20511b1283f4047/src/python/pants/backend/python/goals/export.py#L381

Reply:
Good callout. Have done this.